rust/gitxetcore/src/smudge_query_interface.rs

```diff
@@ -52,6 +55,8 @@ pub struct FileReconstructionInterface {
     pub smudge_query_policy: SmudgeQueryPolicy,
     pub shard_manager: Arc<ShardFileManager>,
     pub shard_client: Option<GrpcShardClient>,
+    pub reconstruction_cache:
+        Mutex<LruCache<merklehash::MerkleHash, (MDBFileInfo, Option<MerkleHash>)>>,
```

Comment:
nit: we can get better performance by using a `std::sync::Mutex` instead of a `tokio::sync::Mutex` since we are not holding the lock across await points ([recommended in tokio docs](https://tokio.rs/tokio/tutorial/shared-state#use-tokios-asynchronous-mutex)).

Reply:
Oh interesting. Good to know.

Reply:
fixed